### ISSUE TITLE
Use more efficient language for backend runtime

### ISSUE THREAD

**Issue by Lynzzyr** [logandhillon/quark#3]
The leading quantum computing SDK/library is Qiskit by IBM. However it is a Python library. There are some undocumented ways to use C++ with Qiskit, but generally it is for Python. Preferably the backend runtime should be coded in a more efficient language such as the aforementioned C++ or even god forbid Rust. While Qiskit isn't available for these languages, here are some other libraries for other languages:

[Qulacs](https://github.com/qulacs/qulacs) - C++ (most popular for C++)
[Quantum++](https://github.com/softwareQinc/qpp) - C++
[RustQIP](https://github.com/Renmusxd/RustQIP) - Rust
[Quantum](https://github.com/beneills/quantum) - Rust

**Comment by logandhillon** [comment]
@bulutthecat Please make sure you are not directly committing to main. I have reverted your commit on main and left your branch alone, however, we want to keep main as the working branch.

When you're done, make a PR instead.

**Comment by Lynzzyr** [comment]
While a compiler written in Rust may be a smidge faster, ultimately a Python compiler would be more practical as it has virtually no performance drawbacks and Rust may in fact be a hinderance in terms of the host machine and the person working on the script.

This issue is meant to actually run the quantum code in another language besides Python. I will be working on this and I may use the Rust compiler or not. Probably not though.

**Comment by bulutthecat** [comment]
> @bulutthecat Please make sure you are not directly committing to main. I have reverted your commit on main and left your branch alone, however, we want to keep main as the working branch.
> 
> When you're done, make a PR instead.

The changes I have pushed to main, I have made sure to not interfere with the main frontend/backend. As well as rigorously tested to make sure no existing code in main was modified.
Please let me know before you make a change like that.

**Comment by bulutthecat** [comment]
> While a compiler written in Rust may be a smidge faster, ultimately a Python compiler would be more practical as it has virtually no performance drawbacks and Rust may in fact be a hinderance in terms of the host machine and the person working on the script.
> 
> This issue is meant to actually run the quantum code in another language besides Python. I will be working on this and I may use the Rust compiler or not. Probably not though.

Yes, currently Python is fine for that, our compiler needs to be well optimized for RAM usage though, as each user query will be going through it.
Python has proven to use alot of RAM, something which cannot be particularly reserved at scale, for each user request, and JIT isnt fast enough for our usecase.

As for moving to Rust or C++ for our compiled code, It would be easier to use native Qiskit instead of a subsidiary/intermediary language.

**Comment by Lynzzyr** [comment]
Okay I will look into your suggestions however note that those mentioned libraries are not intermediary languages for Qiskit and are instead fully fledged quantum SDKs as far as my knowledge covers.

For now @bulutthecat you don't need to worry about this.

**Comment by logandhillon** [comment]
> The changes I have pushed to main, I have made sure to not interfere with the main frontend/backend. As well as rigorously tested to make sure no existing code in main was modified. Please let me know before you make a change like that.

Hi, please be aware that no one should commit directly to main, instead, make a branch and PR into main. I'm sure you're aware of this, so no worries.
